File: Fizzbuzz/c++/try.01/fizzbuzz/try3/FizzBuzz.cpp
#include "FizzBuzz.h"

FizzBuzz::FizzBuzz(void)
{
	magicNumbers.insert(magicNumberEntry(3,"fizz"));
	magicNumbers.insert(magicNumberEntry(5,"buzz"));
	magicNumbers.insert(magicNumberEntry(6,"moo"));
	magicNumbers.insert(magicNumberEntry(15,"fizzbuzz"));
	magicNumbers.insert(magicNumberEntry(33,"fizzyfizz"));
}


FizzBuzz::~FizzBuzz(void)
{
}

void FizzBuzz::GetFizzBuzzList(int count, string* pList)
{
	for(int i = 1; i <= count; i++)
		pList[i-1] = GetFizzBuzzValue(i);
}

string FizzBuzz::GetFizzBuzzValue(int number)
{
	map<int, string>::reverse_iterator iter = magicNumbers.rbegin();
	for(; iter != magicNumbers.rend(); iter++)
		if(number % iter->first == 0)
			return iter->second;

	return to_string(number);
}

File: Android/Fizzbuzz/app/src/main/java/com/tonytroy/fizzbuzz2/Fizzbuzz.java
package com.tonytroy.fizzbuzz2;

/**
 * Created by TT on 6/19/2016.
 */
public class Fizzbuzz {

    public String[] getList(int listSize) {
        String[] list = new String[listSize];
        for (int i = 1; i <= listSize; i++) {
            list[i-1] = GetFizzbuzzValue(i);
        }
        return list;
    }

    private String GetFizzbuzzValue(int i) {
        String value = Integer.toString(i);

        if (i % 15 == 0)
            value = "FizzBuzz";
        else if (i % 5 == 0)
            value = "Buzz";
        else if (i % 3 == 0)
            value = "Fizz";

        return value;
    }
}